cinema/views.py

```diff
@@ -1 +1,85 @@
-# write views here
+from rest_framework import viewsets
+from cinema.models import (
+    CinemaHall,
+    Genre,
+    Actor,
+    Movie,
+    MovieSession,
+    Order,
+    Ticket,
+)
+from cinema.serializers import (
+    ActorSerializer,
+    ActorListSerializer,
+    CinemaHallSerializer,
+    GenreSerializer,
+    MovieSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionRetrieveSerializer,
+    OrderSerializer,
+    TicketSerializer,
+)
+
+
+class CinemaHallSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class GenreSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return ActorListSerializer
+        return ActorSerializer
+
+
+class MovieSet(viewsets.ModelViewSet):
+
+    def get_queryset(self):
+        queryset = Movie.objects.all()
+        if self.action in ["list", "retrieve"]:
+            queryset = queryset.prefetch_related("genres", "actors")
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        if self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
+
+
+class MovieSessionSet(viewsets.ModelViewSet):
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
+        if self.action in ["list", "retrieve"]:
+            queryset = queryset.select_related("movie", "cinema_hall")
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+        if self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+
+class OrderSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+
+
+class TicketSet(viewsets.ModelViewSet):
+    queryset = Ticket.objects.all()
+    serializer_class = TicketSerializer
```

Comment:
All names of Views should contain `view` word inside. Example;

`OrderViewSet`. Fix in all cases